lib/livebook/live_markdown/import.ex

```diff
@@ -393,19 +393,47 @@ defmodule Livebook.LiveMarkdown.Import do
   end
 
   defp postprocess_notebook(notebook) do
-    sections =
-      Enum.map(notebook.sections, fn section ->
+    {sections, {_branching_ids, warnings}} =
+      notebook.sections
+      |> Enum.with_index()
+      |> Enum.map_reduce({MapSet.new(), []}, fn {section, section_idx},
+                                                {branching_ids, warnings} ->
         # Set parent_id based on the persisted branch_parent_index if present
         case section.parent_id do
           nil ->
-            section
+            {section, {branching_ids, warnings}}
 
           {:idx, parent_idx} ->
             parent = Enum.at(notebook.sections, parent_idx)
-            %{section | parent_id: parent.id}
+
+            cond do
+              section_idx < parent_idx ->
```

Comment:
In the original issue a section pointed to itself, we don't want that too :)

```suggestion
              section_idx <= parent_idx ->
```